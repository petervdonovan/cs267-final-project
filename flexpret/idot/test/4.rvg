[mu []
[def (N) 8]
[def (A) {t2}]
[def (B) {t3}]
[def (C) {t4}]
[[file {Reset_Handler} {
  Reset_Handler:
    [id init]
    li A 0
    li B 0
    li C 0
    [single-hart a0 {
      [allocate N A]
      [allocate N B]
      [allocate N C]
      [for t0 t1 0 to [* 4 N] by 4 {
        add a1 A t0
        addi a2 t0 [/ N -8]
        sw a2 0(a1)
      }]
      [for t0 t1 0 to [* 4 N] by 4 {
        add a1 B t0
        li a2 [/ N 2]
        sw a2 0(a1)
      }]
    }]
    [single-core a2 {
      [reduce-shmem +acc A 0 a0 a1]
      [reduce-shmem +acc B 0 a0 a1]
      [reduce-shmem +acc C 0 a0 a1]
    }]
    [single-core a0
      [parfor t0 t1 0 to [* 4 N] by 4 {
        add a1 A t0
        add a2 B t0
        add a3 C t0
        lw a1 0(a1)
        lw a2 0(a2)
        [max~ a4 a1 a2]
        sw a4 0(a3)
      }]]
    [single-core a0
      [single-hart a1
        [for t0 t1 0 to [* 4 N] by 4
        {
          add a3 C t0
          lw a2 0(a3)
          [sim-print-int 42 a4 a5]
          [sim-print a2 a3]
        }]]]
    [single-core a0 [single-hart a1 [sim-print-int 99 a4 a5]]]
    [id die]}]]
]
