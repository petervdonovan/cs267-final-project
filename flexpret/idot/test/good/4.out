Core-0:         42
Core-0:          4
Core-0:         42
Core-0:          4
Core-0:         42
Core-0:          7
Core-0:         42
Core-0:         11
Core-0:         42
Core-0:         15
Core-0:         42
Core-0:         19
Core-0:         42
Core-0:         23
Core-0:         42
Core-0:         27
Core-0:         99
Core-0 is done
Core-1 is done
Core-2 is done
Core-3 is done
Core-4 is done
Core-5 is done
Core-6 is done
Core-7 is done
Core-8 is done
Core-9 is done
Core-10 is done
Core-11 is done
Core-12 is done
Core-13 is done
Core-14 is done
Core-15 is done
All cores are done terminating
Assertion failed: Program terminated sucessfully
